tests/unit/browser/webkit/network/test_pac.py

```diff
@@ -0,0 +1,242 @@
+
+
+def test_myIpAddress():
+    _pac_equality_test("isResolvable(myIpAddress())", "true")
+
+
+def test_dnsDomainLevels():
+    _pac_equality_test("dnsDomainLevels('www')", "0")
+    _pac_equality_test("dnsDomainLevels('www.netscape.com')", "2")
+
+
+def test_shExpMatch():
+    _pac_equality_test("shExpMatch('http://home.netscape.com/people/ari/index.html', '*/ari/*')", "true")
+    _pac_equality_test("shExpMatch('http://home.netscape.com/people/montulli/index.html', '*/ari/*')", "false")
+
+
+def test_weekdayRange(caplog):
+    _pac_equality_test("weekdayRange('MON', 'FRI')", "true")
+    _pac_equality_test("weekdayRange('MON', 'FRI', 'GMT')", "true")
+    _pac_equality_test("weekdayRange('SAT')", "false")
+    _pac_equality_test("weekdayRange('SAT', 'GMT')", "false")
+    _pac_equality_test("weekdayRange('FRI', 'MON')", "false")
+
+    _pac_noexcept_test("weekdayRange('SAT')")
+    _pac_except_test(caplog, "weekdayRange('SAT', 'SAT', 'SAT')")
+
+
+def test_dateRange(caplog):
+    _pac_equality_test("dateRange(26)", "true")
+    _pac_equality_test("dateRange(26, 'GMT')", "true")
+    _pac_equality_test("dateRange(1, 15)", "false")
+    _pac_equality_test("dateRange(24, 'DEC')", "false")
+    _pac_equality_test("dateRange('APR', 'SEP')", "true")
+    _pac_equality_test("dateRange(1, 'JUN', 30, 'AUG')", "true")
+    _pac_equality_test("dateRange(1, 'JUN', 1995, 30, 'AUG', 2016)", "true")
+    _pac_equality_test("dateRange('OCT', 2016, 'MAR', 2017)", "false")
+    _pac_equality_test("dateRange(2016)", "true")
+    _pac_equality_test("dateRange(2016, 2014)", "false")
+
+    _pac_except_test(caplog, "dateRange()")
+    _pac_except_test(caplog, "dateRange('GMT')")
+    _pac_except_test(caplog, "dateRange(2016, 'JUN')")
+    _pac_except_test(caplog, "dateRange(2016, 2016, 2016)")
+
+
+def test_timeRange(caplog):
+    _pac_equality_test("timeRange(10)", "true")
+    _pac_equality_test("timeRange(12, 13)", "false")
+    _pac_equality_test("timeRange(8, 'GMT')", "true")
+    _pac_equality_test("timeRange(9, 17)", "true")
+    _pac_equality_test("timeRange(8, 30, 17, 0, 'GMT')", "true")
+    _pac_equality_test("timeRange(0, 0, 0, 8, 30, 0, 'GMT')", "true")
+
+    _pac_except_test(caplog, "timeRange()")
+    _pac_except_test(caplog, "timeRange(0, 0, 0, 0, 0, 0, 0)")
+
+
+def test_proxyBindings():
+    _pac_equality_test("JSON.stringify(ProxyConfig.bindings)", "'{}'")
+
+
+def test_invalid_port():
+    test_str = """
+        function FindProxyForURL(domain, host) {
+            return "PROXY 127.0.0.1:FOO";
+        }
+    """
+
+    res = pac.PACResolver(test_str)
+    try:
+        res.resolve(QNetworkProxyQuery(QUrl("https://example.com/test")))
+        assert False
+    except pac.ParseProxyError:
+        pass
+
+
+# See https://github.com/The-Compiler/qutebrowser/pull/1891#issuecomment-259222615
+@pytest.mark.skipif(QT_VERSION_STR == "5.7.0",
+                    reason="Segfaults when run with QtWebEngine tests")
```

Comment:
I [have tested](https://github.com/The-Compiler/qutebrowser/pull/1891#issuecomment-262778115) that Qt <5.7 does not trigger that problem. Other checks are okay.